src/main/java/org/apache/commons/collections4/bloomfilter/package-info.java

```diff
@@ -41,30 +41,60 @@
  * {@code add}, and {@code subtract} may throw exceptions. Once an exception is thrown the state of the Bloom filter is unknown.
  * The choice to use not use atomic transactions was made to achieve maximum performance under correct usage.</p>
  *
- * <p>In addition the architecture is designed so that the implementation of the storage of bits is abstracted.
- * Programs that utilize the Bloom filters may use the {@code BitMapProducer} or {@code IndexProducer} to retrieve a
- * representation of the internal structure. Additional methods are available in the {@code BitMap} to assist in
- * manipulation of the representations.</p>
+ * <h4>Nomenclature</h4>
  *
- * <p>The Bloom filter code is an interface that requires implementation of 9 methods:</p>
+ * <ul>
+ *     <li>BitMap - In the bloomfilter package a BitMap is not a structure but a logical construct.  It is conceptualized
+ *     as an ordered collection of {@code long} values each of which is interpreted as a 64-bit bit vector.  The mapping of
+ *     bits into the {@code long} values ss described in the the {@code BitMaps} javadoc.</li>
+ *
+ *     <li>Index - In the bloomfilter package an Index is a logical collection of {@code int}s specifying the enabled
+ *     bits in the BitMap.</li>
+ *
+ *     <li>Cell - Some Bloom filters (e.g. CountingBloomFilter) use counters rather than bits.  In the bloomfilter package
+ *     Cells are pairs of ints representing an index and a value.  They are not {@code Pair} objects.  </li>
```

Comment:
I do not see why we require mentioning Pair. A Cell could be a Pair<Integer, Integer>. But it could be a long packed as upper and lower 32-bits, or some other structure. Dropping the sentence does no harm IMO.